packages/eslint-plugin/docs/rules/promise-function-async.md

```diff
@@ -39,4 +45,14 @@ const arrowFunctionReturnsPromise = async () => Promise.resolve('value');
 async function functionReturnsPromise() {
   return Promise.resolve('value');
 }
+
+function functionReturnsUnionWithPromiseExplicitly(
```

Comment:
I think this bears repeating in the context of an example

```suggestion
// an explicit return type that is not Promise means this function cannot be made async, and is thus ignored by the rule
function functionReturnsUnionWithPromiseExplicitly(
```

Reply:
Sounds great